doc/neps/nep-0029-dtype-as-type.rst

```diff
@@ -0,0 +1,268 @@
+===============================================
+NEP 29 — Refactor Dtypes to become Type Objects
+===============================================
+
+:Author: Matti Picus
+:Status: Draft
+:Type: Standards Track
+:Created: 2018-12-27
+
+
+Abstract
+========
+
+NumPy's `dtype <http://www.numpy.org/devdocs/reference/generated/numpy.dtype.html>`
+is a python class with the simple ``mro`` ``[np.dtype, object]``. Creating an
+instance of ``dtype`` *i.e.* ``a = np.dtype('int8')`` will result in a python
+object of type ``dtype``. The ``dtype`` obect instance has attributes, among
+them ``a.type``, which is a class object. Instantiating that class object
+``a.type(3)`` produces a numpy `scalar
+<http://www.numpy.org/devdocs/reference/arrays.scalars.html>`.
+
+This NEP proposes a different class heirarchy. Objects of ``np.dtype`` will
+become type objects with a heirarchical ``mro`` like scalars. They will support
+subclassing. A future NEP may propose that instantiating a dtype type object
```

Comment:
I think this might need clarification: if I understand correctly, one would subclass a descriptor to create something new, one would generally not subclass `np.dtype` itself, since that is the metaclass.

Reply:
Correct. Is it clearer to say "`ndarray` descriptors will have `np.dtype` as their metaclass with a heirarchical ...". I am not sure how to phrase the idea.